hazelcast/src/main/java/com/hazelcast/jet/config/JetConfig.java

```diff
@@ -37,6 +39,36 @@
     public JetConfig() {
     }
 
+    /**
+     * Returns if Jet is enabled
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Sets if Jet is enabled
+     */
+    public JetConfig setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    /**
+     * Returns if uploading resources when submitting the job enabled
+     */
+    public boolean isUploadResources() {
+        return uploadResources;
+    }
+
+    /**
+     * Sets if uploading resources when submitting the job enabled
```

Comment:
I'm surprised that we enable it for the whole member. What's the point of it? Security? Without resources, you can do very little using DAG or Pipeline API - no grouping, no filtering, no mapping, all these require lambdas. But you can do SQL backed by Jet, that's true. So maybe the use case is if you use Jet only for SQL and want to prevent user code upload. It would be good to write here what it's used for.

It's implemented in a way that we still upload the resources, we just don't load them. We cannot avoid that i guess because the client doesn't know the member's configuration. The name also suggests that it applies to all resources - but it doesn't apply to attachment files (`JobConfig.attachFile()`). But I think we could check it earlier - when the master receives the DAG and config, it will check the config if it has any attachments and reject it right away, even if the resources aren't used.

Reply:
Yes, this is considered as a security feature. I'll update the javadoc.

I'll take a look at the `attachFile` suggestion too

Reply:
Also please add a check on master that there are no resources, not just make the job fail with ClassNotFoundException once they are actually used. The user might notice the mistake earlier.

Reply:
looks like we add a `ResourceConfig` even for `JobConfig#attachFile` so it is handled too.